File: 01017. Convert to Base -2/solution.go
/**
 * Problem: 1017. Convert to Base -2
 *
 * Difficulty: Medium
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func baseNeg2(n int) string {
	// Base case: if n is 0, return "0"
	if n == 0 {
		return "0"
	}

	// Create a slice to store the digits
	var digits []byte

	// Convert n to base -2
	for n != 0 {
		// Calculate remainder when divided by -2
		remainder := n % -2
		n /= -2

		// Handle negative remainder by making it positive
		if remainder < 0 {
			remainder += 2
			n++
		}
		
		// Prepend the digit to the result
		digits = append([]byte{byte(remainder + '0')}, digits...)
	}
	
	// Convert the byte slice to string and return
	return string(digits)
}

File: 01017. Convert to Base -2/solution.js
/**
 * Problem: 1017. Convert to Base -2
 *
 * Difficulty: Medium
 *
 * Language: JavaScript
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

/**
 * Convert an integer to its binary representation in base -2
 *
 * @param {number} n - The integer to convert
 *
 * @returns {string} - The binary string in base -2
 */
const baseNeg2 = (n) => ToNegabinary(BigInt(n))

/**
 * Convert a BigInt to its binary representation in base -2
 *
 * @param {bigint} num - The BigInt to convert
 *
 * @returns {string} - The binary string in base -2
 */
const ToNegabinary = (num) => {
  // Initialize an empty string to store the result
  let binaryStr = ''

  while (num !== 0n) {
    // Calculate remainder when divided by -2
    let remainder = num % -2n
    num /= -2n

    // Handle negative remainder by converting to positive
    if (remainder < 0n) {
      remainder += 2n
      num++
    }

    // Prepend the remainder digit to the result string
    binaryStr = remainder.toString() + binaryStr
  }

  // Return "0" if result is empty, otherwise return the binary string
  return binaryStr.length === 0 ? '0' : binaryStr
}
